### ISSUE TITLE
CAnnot set IP address

### ISSUE THREAD

**Issue by fpoto** [gerbera/gerbera#870]
Trying to set the IP address to 127.0.0.1, either on the command line or in the conf file, crashes the server.

If this is an unknown problem, I can provide logs and whatever.

**Comment by whyman** [comment]
It makes no sense to bind to localhost, your clients will not be able to access the server.

Please post a log file.

**Comment by fpoto** [comment]
> It makes no sense to bind to localhost, your clients will not be able to access the server.

I see, thanks. The docs should be amended. When I read http://docs.gerbera.io/en/latest/config-server.html I find in the <server> section the description of <port>, <ip>, <interface>. The first one is clearly defined as pertaining to the UI, the two subsequent ones are not, so from an external point of view (I am no Gerbera expert) it would naturally follow that they would be associated with the UI.

Moreover, an UIip is sorely missing: most security problems would be simply solved by having the UI only accessible through the localhost loopback interface.

In summary, I see here two issues:
- a documentation bug
- an important wishlist about being able to have the UI only on localhost

I suggest this report to be opened again.

**Comment by fpoto** [comment]
> Please post a log file.

```
May 15 22:22:39 pot systemd[1]: Started Gerbera Media Server with MySQL.
 info: [main.cc:77] logCopyright(): Gerbera UPnP Server version 1.5.0 - http://gerbera.io/
 info: [main.cc:78] logCopyright(): ===============================================================================
 info: [main.cc:79] logCopyright(): Gerbera is free software, covered by the GNU General Public License version 2
 info: [main.cc:80] logCopyright(): Copyright 2016-2020 Gerbera Contributors.
 info: [main.cc:81] logCopyright(): Gerbera is based on MediaTomb: Copyright 2005-2010 Gena Batsyan, Sergey Bostandzhyan, Leonhard Wimmer.
 info: [main.cc:82] logCopyright(): ===============================================================================
 info: [config_manager.cc:155] load(): Loading configuration from: /etc/gerbera/config.xml
 info: [config_manager.cc:162] load(): Checking configuration...
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/tmpdir' using default value: '/tmp/'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/servedir' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/storage/sqlite3/synchronous' using default value: 'off'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/storage/sqlite3/on-error' using default value: 'restore'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/storage/sqlite3/backup/attribute::enabled' using default value: 'no'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/storage/sqlite3/backup/attribute::interval' using default value: '600'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/ui/attribute::poll-when-idle' using default value: 'no'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/ui/attribute::poll-interval' using default value: '2'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/ui/items-per-page/attribute::default' using default value: '25'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/mappings/extension-mimetype/attribute::case-sensitive' using default value: 'no'
debug: [config_manager.cc:539] load(): received UTF-8 from nl_langinfo
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/filesystem-charset' using default value: 'UTF-8'
 info: [config_manager.cc:562] load(): Setting filesystem import charset to UTF-8
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/metadata-charset' using default value: 'UTF-8'
 info: [config_manager.cc:574] load(): Setting metadata import charset to UTF-8
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/playlist-charset' using default value: 'UTF-8'
 info: [config_manager.cc:586] load(): Setting playlist charset to UTF-8
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/interface' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/bookmark' using default value: 'gerbera.html'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/modelName' using default value: 'Gerbera'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/modelDescription' using default value: 'Free UPnP AV MediaServer, GNU GPL'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/modelNumber' using default value: '1.5.0'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/modelURL' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/serialNumber' using default value: '1'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/manufacturer' using default value: 'Gerbera Contributors'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/manufacturerURL' using default value: 'http://gerbera.io/'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/presentationURL' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/presentationURL/attribute::append-to' using default value: 'none'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/upnp-string-limit' using default value: '-1'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/scripting/playlist-script/attribute::create-link' using default value: 'yes'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/autoscan/attribute::use-inotify' using default value: 'auto'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/transcoding/attribute::fetch-buffer-size' using default value: '262144'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/transcoding/attribute::fetch-buffer-fill-size' using default value: '0'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/library-options/libexif/auxdata' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/library-options/exiv2/auxdata' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/library-options/id3/auxdata' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/library-options/ffmpeg/auxdata' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/extended-runtime-options/ffmpegthumbnailer/cache-dir' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/server/extended-runtime-options/ffmpegthumbnailer/cache-dir/attribute::enabled' using default value: 'yes'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/magic-file' using default value: ''
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/online-content/SopCast/attribute::enabled' using default value: 'no'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/online-content/SopCast/attribute::refresh' using default value: '0'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/online-content/SopCast/attribute::purge-after' using default value: '0'
debug: [config_manager.cc:1221] getOption(): Config: option not found: '/config/import/online-content/SopCast/attribute::update-at-start' using default value: 'no'
 info: [config_manager.cc:1200] load(): Configuration check succeeded.
debug: [config_manager.cc:1204] load(): Config file dump after validation: <config version="2" xmlns="http://mediatomb.cc/config/2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://mediatomb.cc/config/2 http://mediatomb.cc/config/2.xsd">

>
>
 show-tooltips="yes">
ed="no" session-timeout="30">
="gerbera" password="gerbera" />


me>
e-4580-4526-88b9-57168c08a8ea</udn>
rbera/.config/gerbera</home>
re/gerbera/web</webroot>
e>

d="yes">
e>gerbera.db</database-file>

"no">
st</host>
bera</username>
bera</database>


-options>
ler enabled="yes">
ze>128</thumbnail-size>
age>5</seek-percentage>
erlay>yes</filmstrip-overlay>
ugs>no</workaround-bugs>
y>8</image-quality>
iler>
ems enabled="no" suppress-cds-updates="yes">
"prepend">*</string>

deo</content>

tems>
e-options>

s="no">
-charset="UTF-8">
/usr/share/gerbera/js/common.js</common-script>
t>/usr/share/gerbera/js/playlists.js</playlist-script>
 type="builtin">
t>/usr/share/gerbera/js/import.js</import-script>
t>


type ignore-unknown="no">
3" to="audio/mpeg" />
x" to="application/ogg" />
v" to="video/ogg" />
a" to="audio/ogg" />
g" to="audio/ogg" />
m" to="video/ogg" />
f" to="video/x-ms-asf" />
x" to="video/x-ms-asf" />
a" to="audio/x-ms-wma" />
x" to="audio/x-ms-wax" />
v" to="video/x-ms-wmv" />
x" to="video/x-ms-wvx" />
" to="video/x-ms-wm" />
x" to="video/x-ms-wmx" />
u" to="audio/x-mpegurl" />
s" to="audio/x-scpls" />
v" to="video/x-flv" />
v" to="video/x-matroska" />
a" to="audio/x-matroska" />
f" to="audio/x-dsd" />
f" to="audio/x-dsd" />
" to="audio/x-wavpack" />
etype>
lass>
dio/*" to="object.item.audioItem.musicTrack" />
deo/*" to="object.item.videoItem" />
age/*" to="object.item.imageItem" />
plication/ogg" to="object.item.audioItem.musicTrack" />
class>
nttype>
pe="audio/mpeg" as="mp3" />
pe="application/ogg" as="ogg" />
pe="audio/ogg" as="ogg" />
pe="audio/x-flac" as="flac" />
pe="audio/flac" as="flac" />
pe="audio/x-ms-wma" as="wma" />
pe="audio/x-wavpack" as="wv" />
pe="image/jpeg" as="jpg" />
pe="audio/x-mpegurl" as="playlist" />
pe="audio/x-scpls" as="playlist" />
pe="audio/x-wav" as="pcm" />
pe="audio/L16" as="pcm" />
pe="video/x-msvideo" as="avi" />
pe="video/mp4" as="mp4" />
pe="audio/mp4" as="mp4" />
pe="video/x-matroska" as="mkv" />
pe="audio/x-matroska" as="mka" />
pe="audio/x-dsd" as="dsd" />
enttype>


enabled="no" refresh="43200" update-at-start="no" resolution="640" />


d="yes">
-mappings>
type="video/x-matroska" using="ffmpegavi2h264" />
type="video/x-flv" using="vlcmpeg" />
type="application/ogg" using="vlcmpeg" />
type="audio/ogg" using="ogg2mp3" />
e-mappings>

ogg2mp3" enabled="yes" type="external">
io/mpeg</mimetype>
o</accept-url>
ce>yes</first-resource>
heora>no</accept-ogg-theora>
d="ffmpeg" arguments="-y -i %in -f mp3 %out" />
"1048576" chunk-size="131072" fill-size="262144" />

vlcmpeg" enabled="yes" type="external">
eo/mpeg</mimetype>
es</accept-url>
ce>yes</first-resource>
heora>yes</accept-ogg-theora>
d="vlc" arguments="-I dummy %in --sout #transcode{venc=ffmpeg,vcodec=mp2v,vb=4096,fps=25,aenc=ffmpeg,acodec=mpga,ab=192,samplerate=44100,channels=2}:standard{access=file,mux=ps,dst=%out} vlc:quit" />
"14400000" chunk-size="512000" fill-size="120000" />

ffmpegavi2h264" enabled="yes" type="external">
eo/x-matroska</mimetype>
o</accept-url>
ce>yes</first-resource>
d="ffmpeg" arguments="-y -i %in -c:v libx264 -preset fast -c:a copy %out" />
"4194304" chunk-size="524288" fill-size="1048576" />




debug: [config_manager.cc:1704] dumpOptions(): Dumping options!
debug: [config_manager.cc:1713] dumpOptions():  IntOption 00 49452
debug: [config_manager.cc:1708] dumpOptions():     Option 01 127.0.0.1
debug: [config_manager.cc:1708] dumpOptions():     Option 02
debug: [config_manager.cc:1708] dumpOptions():     Option 03 Gerbera
debug: [config_manager.cc:1708] dumpOptions():     Option 04 Gerbera Contributors
debug: [config_manager.cc:1708] dumpOptions():     Option 05 http://gerbera.io/
debug: [config_manager.cc:1708] dumpOptions():     Option 06 Gerbera
debug: [config_manager.cc:1708] dumpOptions():     Option 07 Free UPnP AV MediaServer, GNU GPL
debug: [config_manager.cc:1708] dumpOptions():     Option 08 1.5.0
debug: [config_manager.cc:1708] dumpOptions():     Option 09
debug: [config_manager.cc:1708] dumpOptions():     Option 10 1
debug: [config_manager.cc:1708] dumpOptions():     Option 11
debug: [config_manager.cc:1708] dumpOptions():     Option 12 none
debug: [config_manager.cc:1708] dumpOptions():     Option 13 uuid:5c98862e-4580-4526-88b9-57168c08a8ea
debug: [config_manager.cc:1708] dumpOptions():     Option 14 /var/lib/gerbera/.config/gerbera
debug: [config_manager.cc:1708] dumpOptions():     Option 15 /tmp/
debug: [config_manager.cc:1708] dumpOptions():     Option 16 /usr/share/gerbera/web
debug: [config_manager.cc:1708] dumpOptions():     Option 17 /var/lib/gerbera/.config/gerbera/
debug: [config_manager.cc:1713] dumpOptions():  IntOption 18 1800
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 19 false
debug: [config_manager.cc:1708] dumpOptions():     Option 20 /var/lib/gerbera/.config/gerbera/gerbera.html
debug: [config_manager.cc:1713] dumpOptions():  IntOption 21 -1
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 22 true
debug: [config_manager.cc:1713] dumpOptions():  IntOption 23 2
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 24 false
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 25 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 27 30
debug: [config_manager.cc:1713] dumpOptions():  IntOption 28 25
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 30 true
debug: [config_manager.cc:1708] dumpOptions():     Option 31 sqlite3
debug: [config_manager.cc:1708] dumpOptions():     Option 32 /var/lib/gerbera/.config/gerbera/gerbera.db
debug: [config_manager.cc:1713] dumpOptions():  IntOption 33 0
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 34 true
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 35 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 36 600
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 43 true
debug: [config_manager.cc:1713] dumpOptions():  IntOption 44 128
debug: [config_manager.cc:1713] dumpOptions():  IntOption 45 5
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 46 true
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 47 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 48 8
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 49 true
debug: [config_manager.cc:1708] dumpOptions():     Option 50
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 51 false
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 52 true
debug: [config_manager.cc:1708] dumpOptions():     Option 53 *
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 54 true
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 56 false
debug: [config_manager.cc:1708] dumpOptions():     Option 57 UTF-8
debug: [config_manager.cc:1708] dumpOptions():     Option 58 UTF-8
debug: [config_manager.cc:1708] dumpOptions():     Option 59 UTF-8
debug: [config_manager.cc:1708] dumpOptions():     Option 60 UTF-8
debug: [config_manager.cc:1708] dumpOptions():     Option 61 /usr/share/gerbera/js/common.js
debug: [config_manager.cc:1708] dumpOptions():     Option 62 /usr/share/gerbera/js/playlists.js
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 63 true
debug: [config_manager.cc:1708] dumpOptions():     Option 64 /usr/share/gerbera/js/import.js
debug: [config_manager.cc:1708] dumpOptions():     Option 65 builtin
debug: [config_manager.cc:1708] dumpOptions():     Option 66
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 68 true
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 70 false
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 71 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 79 262144
debug: [config_manager.cc:1713] dumpOptions():  IntOption 80 0
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 81 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 82 0
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 83 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 84 0
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 85 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 86 43200
debug: [config_manager.cc:1718] dumpOptions(): BoolOption 87 false
debug: [config_manager.cc:1713] dumpOptions():  IntOption 88 43200
debug: [config_manager.cc:1708] dumpOptions():     Option 89 640
debug: [sqlite3_storage.cc:299] exec(): Adding query to Queue: PRAGMA locking_mode = EXCLUSIVE
debug: [sqlite3_storage.cc:536] run(): PRAGMA locking_mode = EXCLUSIVE
debug: [sqlite3_storage.cc:299] exec(): Adding query to Queue: PRAGMA foreign_keys = ON
debug: [sqlite3_storage.cc:536] run(): PRAGMA foreign_keys = ON
debug: [sqlite3_storage.cc:299] exec(): Adding query to Queue: PRAGMA synchronous = 0
debug: [sqlite3_storage.cc:536] run(): PRAGMA synchronous = 0
debug: [sqlite3_storage.cc:204] init(): db_version: 5
debug: [sql_storage.cc:2102] loadLastID(): LoadedId: 71
debug: [sql_storage.cc:2281] doMetadataMigration(): Checking if metadata migration is required
debug: [sql_storage.cc:2289] doMetadataMigration(): mt_cds_object rows having metadata: 0
debug: [sql_storage.cc:2296] doMetadataMigration(): mt_metadata rows having metadata: 10
 info: [sql_storage.cc:2303] doMetadataMigration(): About to migrate metadata from mt_cds_object to mt_metadata
 info: [sql_storage.cc:2304] doMetadataMigration(): No data will be removed from mt_cds_object
 info: [sql_storage.cc:2320] doMetadataMigration(): Migrated metadata - object count: 0
debug: [sql_storage.cc:1616] updateAutoscanPersistentList(): setting persistent autoscans untouched - scanmode: timed;
debug: [sqlite3_storage.cc:299] exec(): Adding query to Queue: UPDATE "mt_autoscan" SET "touched"=0 WHERE "persistent"=1 AND "scan_mode"='timed'
debug: [sqlite3_storage.cc:536] run(): UPDATE "mt_autoscan" SET "touched"=0 WHERE "persistent"=1 AND "scan_mode"='timed'
debug: [sql_storage.cc:1627] updateAutoscanPersistentList(): updating/adding persistent autoscans (count: 0)
debug: [sqlite3_storage.cc:299] exec(): Adding query to Queue: DELETE FROM "mt_autoscan" WHERE "touched"=0 AND "scan_mode"='timed'
debug: [sqlite3_storage.cc:536] run(): DELETE FROM "mt_autoscan" WHERE "touched"=0 AND "scan_mode"='timed'
debug: [server.cc:103] run(): start
debug: [tools.cc:957] ipToInterface(): Looking for '127.0.0.1'
debug: [server.cc:119] run(): Initialising libupnp with interface: 'lo', port: 49452
error: [main.cc:332] main(): LibUPnP error code: -121
debug: [server.cc:262] emptyBookmark(): Clearing bookmark file at: /var/lib/gerbera/.config/gerbera/gerbera.html
debug: [server.cc:278] shutdown(): Server shutting down
error: [server.cc:282] shutdown(): UpnpUnRegisterClient failed (-116)
error: [server.cc:287] shutdown(): UpnpUnRegisterRootDevice failed (-116)
debug: [server.cc:294] shutdown(): now calling upnp finish
debug: [content_manager.cc:332] shutdown(): start
debug: [content_manager.cc:334] shutdown(): updating last_modified data for autoscan in database...
May 15 22:22:39 pot systemd[1]: gerbera.service: Main process exited, code=killed, status=11/SEGV
May 15 22:22:39 pot systemd[1]: gerbera.service: Failed with result 'signal'.
```
